### DISCUSSION TITLE
split a tuple with id + items

### DISCUSSION THREAD

**Discussion by rossijulieta** [nextflow-io/nextflow#3785] [Q&A] [answered]
Hi all, 

I tried to reduce my problem to the minimum in the following example; I have a channel like ch1 with a number representing an id and then a list of values and need to split it so that it looks like the expected values as commented bellow (id + each value).

nextflow.enable.dsl=2
ch1 = Channel.from([1, ['a','b','c']], [2, ['y','z']])

// expected -> [1,'a'], [1,'b'],[1,'c'],[2,'y'],[2,'z']

Any hint would be appreciated. 
Thanks

**Comment by mribeirodantas** [comment, marked as answer]
The `transpose` operator does what you're looking for.
```
ch1 = Channel.from([1, ['a','b','c']], [2, ['y','z']])
ch1.transpose().view()
```

Output:

```
N E X T F L O W  ~  version 23.02.0-edge
Launching `a.nf` [sick_jones] DSL2 - revision: 4d0c00fd07
[1, a]
[1, b]
[1, c]
[2, y]
[2, z]
```

**Reply by rossijulieta** [reply]
Excellent! Thanks

**Reply by mribeirodantas** [reply]
You're welcome 😉
